ddp-workspace/projects/ddp-sdk/src/lib/components/changeLanguageRedirect.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+import { StudyLanguage } from "../models/studyLanguage";
+import { ConfigurationService } from "../services/configuration.service";
+import { LanguageService} from "../services/languageService.service";
+import { LanguageServiceAgent } from "../services/serviceAgents/languageServiceAgent.service";
+
+@Component({
+  selector: 'ddp-change-language-redirect',
+  template: `<ng-container></ng-container>`
+})
+export class ChangeLanguageRedirectComponent implements OnInit {
+  private lang: string = null;
+  private supportedLanguages: StudyLanguage[] = null;
+
+  //The destination to redirect to, relative to the main site
+  private dest: string = null;
+
+  //Optionally, a parameter to pass through to the destination to redirect to
+  private queryParamValue: string = null;
+
+  constructor(
+    private languageService: LanguageService,
+    private languageServiceAgent: LanguageServiceAgent,
+    private route: ActivatedRoute,
+    private router: Router,
+    @Inject('ddp.config') private config: ConfigurationService) {
+  }
+
+  public ngOnInit(): void {
+    //Get the specified language and specified destination and store for later
+    let paramPromise: Promise<void> = this.getParamInfo();
+
+    //Get any parameters to pass through to specified destination
+    let queryParamPromise: Promise<void> = this.getQueryParamInfo();
+
+    //Get the study's configured languages and store for later
+    let supportedLanguagesPromise: Promise<void> = this.getSupportedLanguagesPromise();
+
+    //Attempt to change language and redirect
+    Promise.all([paramPromise, queryParamPromise, supportedLanguagesPromise])
```

Comment:
For what reason did you decide to user Promises instead of RxJS? :)
It stands out a bit from the rest SDK's components.

Reply:
I was having trouble chaining together the observables to make sure the app got the supported languages, the parameters, and the query parameters and only after all of that was done, did everything else.  It seems like there must be an RxJS way to do that but I spent enough time trying to figure it out with no luck that I decided to try a different approach.